src/runtime/hashmap.go

```diff
@@ -158,6 +195,32 @@ func hashmapInsertIntoNewBucket(m *hashmap, key, value unsafe.Pointer, tophash u
 	return bucket
 }
 
+func hashmapGrow(m *hashmap, keyEqual func(x, y unsafe.Pointer, n uintptr) bool, keyHash func(x unsafe.Pointer, size uintptr) uint32) {
+
+	// clone map as empty
+	n := *m
+	n.count = 0
+
+	// allocate our new buckets twice as big
+	n.bucketBits = m.bucketBits + 1
+	numBuckets := uintptr(1) << n.bucketBits
+	bucketBufSize := unsafe.Sizeof(hashmapBucket{}) + uintptr(m.keySize)*8 + uintptr(m.valueSize)*8
+	n.buckets = alloc(bucketBufSize*numBuckets, nil)
+
+	// use a hashmap iterator to go through the old map
+	var it hashmapIterator
+
+	var key = alloc(uintptr(m.keySize), nil)
+	var value = alloc(uintptr(m.valueSize), nil)
```

Comment:
You could perhaps merge these two allocations:

```suggestion
	var key = alloc(uintptr(m.keySize)+uintptr(m.valueSize), nil)
	var value = unsafe.Pointer(uintptr(key) + uintptr(m.keySize))
```
(...although this might need some more alignment to be correct)